src/Microsoft.AspNet.Mvc.TagHelpers/AnchorTagHelper.cs

```diff
@@ -32,9 +32,16 @@ public class AnchorTagHelper : TagHelper
         private const string RouteValuesPrefix = "asp-route-";
         private const string Href = "href";
 
-        [Activate]
-        [HtmlAttributeNotBound]
-        public IHtmlGenerator Generator { get; set; }
+        /// <summary>
+        /// Creates a new <see cref="AnchorTagHelper"/>.
+        /// </summary>
+        /// <param name="generator">The <see cref="IHtmlGenerator"/>.</param>
+        public AnchorTagHelper(IHtmlGenerator generator)
+        {
+            Generator = generator;
+        }
+
+        public IHtmlGenerator Generator { get; }
```

Comment:
Not for this PR but there's no reason to use `public` properties for most things in a tag helper now. `private` fields are sufficient. Please file a follow-up bug to clean these out.